heat/core/manipulations.py

```diff
@@ -558,6 +560,49 @@ def expand_dims(a, axis):
     )
 
 
+def flatten(a):
+    """
+    Flattens an array into one dimension
+
+    Parameters
+    ----------
+    a : DNDarray
+        array to collapse
+    Returns
+    -------
+    ret : DNDarray
+        flattened copy
+    Examples
+    --------
+    >>> a = ht.array([[[1,2],[3,4]],[[5,6],[7,8]]])
+    >>> ht.flatten(a)
+    tensor([1,2,3,4,5,6,7,8])
+    """
+    if a.split is None:
+        return factories.array(
+            torch.flatten(a._DNDarray__array),
+            dtype=a.dtype,
+            is_split=None,
+            device=a.device,
+            comm=a.comm,
+        )
+
+    # The resplit function scramble the tensor when switching axes, see issue 425
+    if a.split > 0:
+        a = resplit(a, 0)
+        warnings.warn("The flattened tensor may have a wrong order for split axes > 0", UserWarning)
+    a = factories.array(
+        torch.flatten(a._DNDarray__array),
+        dtype=a.dtype,
+        is_split=a.split,
+        device=a.device,
+        comm=a.comm,
+    )
+    a.balance_()
```

Comment:
replit() + balance_() is a very expensive way of doing flattening. It entails superfluous communication that is then undone again by the next communication. Can we get this working in one go?